apps/ledger-live-mobile/src/components/RootDrawer/RootDrawer.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react";
+
+import { EthereumStakingDrawer } from "../../families/ethereum/EthereumStakingDrawer";
+
+import { RootDrawerProvider, useRootDrawerContext } from "../../context/RootDrawerContext";
+import { InitialDrawerID, RootDrawerProps } from "./types";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { useNavigation } from "@react-navigation/native";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { NavigatorName } from "../../const";
+import { PTXServicesAppleWarning } from "./InitialDrawers/PTXServicesAppleWarning";
+export async function getInitialDrawersToShow(initialDrawers: InitialDrawerID[]) {
+  const initialDrawersToShow = await AsyncStorage.multiGet(initialDrawers);
+
+  // if we have a value then the drawer should not be shown
+  return initialDrawersToShow
+    .map(([key, value]) => (value ? undefined : key))
+    .filter((drawer): drawer is InitialDrawerID => !!drawer);
+}
+
+export function RootDrawerSelector() {
+  const { drawer } = useRootDrawerContext();
+  switch (drawer.id) {
+    case "EthStakingDrawer":
+      return <EthereumStakingDrawer />;
+    case InitialDrawerID.PTXServicesAppleDrawerKey:
+      return <PTXServicesAppleWarning />;
+    default:
+      return null;
+  }
+}
+
+export function RootDrawer({ drawer }: RootDrawerProps) {
+  // this keeps track initial drawers closed during the session.
+  const [initialDrawers, setInitialDrawers] = useState(Object.values(InitialDrawerID));
+  const { navigate } = useNavigation<StackNavigationProp<Record<string, object | undefined>>>();
+
+  useEffect(() => {
+    if (!drawer) {
+      getInitialDrawersToShow(initialDrawers).then(([id]) => {
```

Comment:
would be clearer with an async func here.  You can make it asyncy by wrapping it all in an async func and then calling it, eg:
```
useEffect(() => {
async function displayDrawers(){ ....};
displayDrawers()
},[....])
```